tiledb/rest/curl/curl.cc

```diff
@@ -120,69 +156,44 @@ CURLcode curl_fetch_url(
      * break */
     if (rcode == CURLE_OK)
       break;
-    /* If we are here, then curl returned an error */
-    /* Free allocated memory for return as we are now going to retry */
-    free(fetch->memory);
-    fetch->memory = nullptr;
-    fetch->size = 0;
+
+    /* Retry on curl errors */
   }
 
-  /* return */
   return rcode;
-  STATS_FUNC_OUT(serialization_curl_fetch_url);
-}
 
-tiledb::sm::Status set_auth(
-    CURL* curl, tiledb::sm::Config* config, struct curl_slist** headers) {
-  const char* token = nullptr;
-  if (config->get("rest.token", &token).ok()) {
-    *headers = curl_slist_append(
-        *headers, (std::string("X-TILEDB-REST-API-Key: ") + token).c_str());
-  } else {
-    // Get username
-    const char* username = nullptr;
-    auto st = config->get("rest.username", &username);
-    if (!st.ok())
-      return st;
-    // Get password
-    const char* password = nullptr;
-    st = config->get("rest.password", &password);
-    if (!st.ok())
-      return st;
-    std::string basic_auth = username + std::string(":") + password;
-    curl_easy_setopt(curl, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
-    curl_easy_setopt(curl, CURLOPT_USERPWD, basic_auth.c_str());
-  }
-  return tiledb::sm::Status::Ok();
+  STATS_FUNC_OUT(serialization_curl_fetch_url);
 }
 
 tiledb::sm::Status post_data(
     CURL* curl,
-    tiledb::sm::Config* config,
-    std::string url,
+    const tiledb::sm::Config* config,
+    const std::string& url,
     tiledb::sm::SerializationType serialization_type,
-    MemoryStruct* data,
-    MemoryStruct* returned_data) {
+    tiledb::sm::Buffer* data,
+    tiledb::sm::Buffer* returned_data) {
   STATS_FUNC_IN(serialization_post_data);
 
-  struct curl_slist* headers = nullptr;
-  // Set auth for server
-  auto st = set_auth(curl, config, &headers);
-  if (!st.ok())
-    return st;
-
-  /* HTTP PUT please */
-  curl_easy_setopt(curl, CURLOPT_POST, 1L);
-
   // TODO: If you post more than 2GB, use CURLOPT_POSTFIELDSIZE_LARGE.
-  curl_easy_setopt(curl, CURLOPT_POSTFIELDS, data->memory);
-  curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, data->size);
+  const uint64_t post_size_limit = uint64_t(2) * 1024 * 1024 * 1024;
```

Comment:
We might want to open a new issues for tracking this. We probably want to switch: https://curl.haxx.se/libcurl/c/CURLOPT_POSTFIELDSIZE_LARGE.html